autogpt_platform/backend/backend/blocks/http.py

```diff
@@ -68,20 +69,37 @@ def run(self, input_data: Input, **kwargs) -> BlockOutput:
                     # we should send it as plain text instead
                     input_data.json_format = False
 
-        response = requests.request(
-            input_data.method.value,
-            input_data.url,
-            headers=input_data.headers,
-            json=body if input_data.json_format else None,
-            data=body if not input_data.json_format else None,
-        )
-        result = response.json() if input_data.json_format else response.text
+        try:
+            response = requests.request(
+                input_data.method.value,
+                input_data.url,
+                headers=input_data.headers,
+                json=body if input_data.json_format else None,
+                data=body if not input_data.json_format else None,
+            )
+            result = response.json() if input_data.json_format else response.text
+
+            if response.status_code // 100 == 2:
+                yield "response", result
+            elif response.status_code // 100 == 4:
+                yield "client_error", result
+            elif response.status_code // 100 == 5:
+                yield "server_error", result
+            else:
+                raise ValueError(f"Unexpected status code: {response.status_code}")
```

Comment:
add a log statement with more details, otherwise it'll be impossible to debug if this ever occurs